cmd/gaia/app/sim_test.go

```diff
@@ -67,13 +67,40 @@ func appStateFn(r *rand.Rand, accs []simulation.Account) json.RawMessage {
 		"\t{amount of steak per account: %v, initially bonded validators: %v}\n",
 		amount, numInitiallyBonded)
 
-	// Randomly generate some genesis accounts
-	for _, acc := range accs {
+	// randomly generate some genesis accounts
+	for i, acc := range accs {
 		coins := sdk.Coins{sdk.NewCoin(stakingTypes.DefaultBondDenom, sdk.NewInt(amount))}
-		genesisAccounts = append(genesisAccounts, GenesisAccount{
-			Address: acc.Address,
-			Coins:   coins,
-		})
+		bacc := auth.NewBaseAccountWithAddress(acc.Address)
+		bacc.SetCoins(coins)
+
+		var gacc GenesisAccount
+
+		// create a vesting account under the following conditions:
+		// - not an initially bonded (genesis) validator
+		// - every 10th account past the initial bonded validator set
+		if int64(i) > numInitiallyBonded && i+1%10 == 0 {
```

Comment:
Can we use a random condition instead (each account has some chance of being vesting)?

Reply:
Also there's no reason the initially bonded validators can't have vesting accounts.

Reply:
> Can we use a random condition instead (each account has some chance of being vesting)?

Sure, how do you suggest we do so?

> Also there's no reason the initially bonded validators can't have vesting accounts.

 True, but this caused me headache because the vesting accounts won't have `DelegatedVesting` populated and thus future delegations/undelegations will fail. I _think_ if we want to support this, we have to add such fields to vesting which is a bit ugly imho. Perhaps we should be okay with vesting accnts never being in the initial bonded set.